build.rs

```diff
@@ -1,13 +1,24 @@
 use semver::Version;
-use std::{env, error::Error, fs, path::PathBuf, process::Command};
+use std::{env, error::Error, fs, path::Path, path::PathBuf, process::Command};
 
 fn main() -> Result<(), Box<dyn Error>> {
     // Put the linker script somewhere the linker can find it
     let out = &PathBuf::from(env::var("OUT_DIR")?);
     let mut linker_script = fs::read_to_string("defmt.x.in")?;
-    let output = Command::new("git").args(&["rev-parse", "HEAD"]).output()?;
-    let version = if output.status.success() {
-        String::from_utf8(output.stdout).unwrap()
+    let hash = Command::new("git")
+        .args(&["rev-parse", "HEAD"])
+        .output()
```

Comment:
The `output()` call itself will fail if the program cannot be found, so I don't think the fix can use `.map`

Reply:
the original was not checking the .git path if git was not installed but the new version should